### PR TITLE
Fix to not remove Hash (#) from URL from bug #202

### PR DESCRIPTION
**Pull request by L3O**
Adding ability to use URL arguments if passed, or use the ones already defined in the URL.
Hash now does not get removed from the URL.

### CODE REVIEW

**Comment by codecounselor** [comment]
Thanks for the PR! There is actually a very similar PR sitting open (without tests).  Would you mind reconciling your changes with the ones in #206.
I'm curious why that contributor just removed it all.

If you can add some accompanying unit tests for the behavior you are expecting I'll happily merge this in.

Edit: saw you just closed this (not sure why) but I'm adding my comment anyway.

**Comment by L3O** [comment]
@codecounselor just reopened the PR adding new unit tests.

File: lib/args.js [hunk 1]
```diff
@@ -13,17 +13,17 @@ function urlWithArgs (urlOrFile, args) {
   args = encode(args)
 
   var u
+  var urlData = url.parse(urlOrFile)
   if (urlOrFile.indexOf('http') === 0) {
```
Comment (fraserxu) [inline lib/args.js]: Is `url.protocol` something might be useful here? https://nodejs.org/api/url.html#url_url_protocol

Comment (codecounselor) [reply lib/args.js]: that is probably for the best.  @L3O do you want to change this before I merge it?

Comment (L3O) [reply lib/args.js]: Done.

**Pull request merged**